openapi_operations_test.go

```diff
@@ -61,6 +61,17 @@ func TestHeaderParams(t *testing.T) {
 	require.Equal(t, "my description", route.Operation.Parameters.GetByInAndName("header", "my-header").Description)
 }
 
+func TestCustomError(t *testing.T) {
+	type MyError struct {
+		Message string
+	}
+	s := NewServer()
+	route := Get(s, "/test", testController).
+		AddError(400, "My Validation Error", MyError{})
+
+	require.Equal(t, "My Validation Error", *route.Operation.Responses.Map()["400"].Value.Description)
+}
+
 func TestCookieParams(t *testing.T) {
```

Comment:
Mind adding a test showing the behavior when Errors are set globally and on a route? Also display what should happen when status code is declared globally but then overridden by the route. I think this would be good too make sure we don't make a breaking change in the future

Reply:
Very good comment indeed, the tests should reflect this feature's behaviour and show some examples!

Adding this right now.